staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_progress.go

```diff
@@ -95,15 +95,16 @@ func (pr *conditionalProgressRequester) Run(stopCh <-chan struct{}) {
 
 		select {
 		case <-timer.C():
-			timer.Reset(progressRequestPeriod)
 			shouldRequest := func() bool {
 				pr.mux.Lock()
 				defer pr.mux.Unlock()
 				return pr.waiting > 0 && !pr.stopped
 			}()
 			if !shouldRequest {
+				timer.Reset(0)
```

Comment:
To double check - Reset(0) will schedule the next tick immediately, so that we will fire immediately when the new waiting reader will be registered, right?

Reply:
Based on the code yes, calling `Reset(0)` will result in `when(0)` returning current time, see https://go.dev/src/time/sleep.go.

Will run the scalability tests again to confirm.

Reply:
Can we add some simple unit test for this usecase?

Reply:
Will try, however after the accident with `timer` vs `ticker` I no longer trust k8s fake implementation to match behavior of real time package.

Reply:
Got 395ms max latency with the change, larger than I would expect before it. Will need to investigate it more.

Reply:
Run the test second time, looks better. There was request with latency of 340ms, but it was not caused by progress notifier. 

Works as predicted